discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -38,21 +38,26 @@ final class ServiceDiscovery(implicit system: ExtendedActorSystem) extends Exten
         .createInstanceFor[SimpleServiceDiscovery](clazzName, (classOf[ExtendedActorSystem] → system) :: Nil)
         .recoverWith {
           case _ ⇒
-            dynamic.createInstanceFor[SimpleServiceDiscovery](_simpleImplMethod,
-              (classOf[ActorSystem] → system) :: Nil)
+            dynamic.createInstanceFor[SimpleServiceDiscovery](clazzName, (classOf[ActorSystem] → system) :: Nil)
         }
         .recoverWith {
           case _ ⇒
-            dynamic.createInstanceFor[SimpleServiceDiscovery](_simpleImplMethod, Nil)
+            dynamic.createInstanceFor[SimpleServiceDiscovery](clazzName, Nil)
         }
 
-    val i = create("akka.discovery." + _simpleImplMethod + ".class").recoverWith {
-      case _ ⇒ create(_simpleImplMethod + ".class")
-    }.recoverWith { case _ ⇒ create(_simpleImplMethod) }
+    // format: OFF
+    val i = create {
+      classNameFromConfig("akka.discovery." + _simpleImplMethod + ".class")
+    }.recoverWith {
+      case _ ⇒ create(classNameFromConfig(_simpleImplMethod + ".class"))
+    }.recoverWith {
+      case _ ⇒ create(_simpleImplMethod) // so perhaps, it is a classname?
+    }
+    // format: ON
 
     i.getOrElse(
       throw new IllegalArgumentException(
-          s"Illegal `akka.discovery.impl` value (${_simpleImplMethod}) or incompatible class! " +
+          s"Illegal `akka.discovery.method` value '${_simpleImplMethod}' or incompatible class! " +
```

Comment:
Confused me for a sec that you might expect a method (rather than a class) here, but makes sense if documented I guess :)

Reply:
Yeah, documented as:

```
  # Users MUST configure this value to set the default discovery mechanism.
  #
  # The value can be an implementation config path name, such as "akka-dns",
  # which would attempt to resolve as `akka.discovery.akka-dns` which is expected
  # to contain a `class` setting. As fallback, the root `akka-dns` setting scope
  # would be used. If none of those contained a `class` setting, then the value is
  # assumed to be a class name, and an attempt is made to intanciate it.
  method = "<method>"
```

Reply:
I might be too late to the bike-shedding party, but it would be more consistent to name this `provider`, as we have for [`actor.provider`](https://github.com/akka/akka/blob/master/akka-actor/src/main/resources/reference.conf#L98) which can be either a short name or a FQCN.

Reply:
Provider I wanted to keep for "provides the routes", this is a "discovery method". But yeah, let's not change now I think..